lib/TPP/Transforms/ToBlockLayoutAndBack.cpp

```diff
@@ -964,8 +943,7 @@ void mlir::tpp::populateSimplifyPacking(RewritePatternSet &patterns) {
   tensor::ParallelInsertSliceOp::getCanonicalizationPatterns(patterns, ctx);
   ctx->getLoadedDialect<tensor::TensorDialect>()->getCanonicalizationPatterns(
       patterns);
-  patterns.add<PackOfReshape, FoldExpandShapeInParallelInsertOp,
-               FoldCollapseShapeInExtractSliceOp, FoldUnPackIntoInsertSlice,
-               ForAllIterArgsFolder>(ctx);
+  patterns.add<PackOfReshape, FoldCollapseShapeInExtractSliceOp,
```

Comment:
You're not adding an upstream pass here. Does the upstream pattern runs automatically or was already added?

Reply:
The replacement pattern is part of `tensor::populateReassociativeReshapeFoldingPatterns` which is already present here.